port/common/omrstr.c

```diff
@@ -2195,11 +2199,24 @@ omrstr_current_time_zone(struct OMRPortLibrary *portLibrary, int32_t *secondsEas
 		if (NULL != secondsEast) {
 			*secondsEast = zoneSecondsEast;
 		}
-		if (NULL != zoneNameBuffer) {
-			strncpy(zoneNameBuffer, zoneName, zoneNameBufferLen);
-			if (0 != zoneNameBufferLen) {
-				zoneNameBuffer[zoneNameBufferLen - 1] = '\0';
+		if ((NULL != zoneNameBuffer) && (NULL != zoneName)) {
+			size_t fullLength = strlen(zoneName);
+			size_t trimLength = (fullLength < zoneNameBufferLen) ? fullLength : (zoneNameBufferLen - 1);
+#if defined(J9ZOS390) && !defined(OMR_EBCDIC)
+			char *finalZoneName = e2a(zoneName, trimLength);
+#else /* defined(J9ZOS390) && !defined(OMR_EBCDIC) */
+			const char *finalZoneName = zoneName;
+#endif /* defined(J9ZOS390) && !defined(OMR_EBCDIC) */
+
+			if (NULL == finalZoneName) {
+				trimLength = 0;
+			} else {
+				memcpy(zoneNameBuffer, finalZoneName, trimLength);
+#if defined(J9ZOS390) && !defined(OMR_EBCDIC)
+				free(finalZoneName);
+#endif /* defined(J9ZOS390) && !defined(OMR_EBCDIC) */
 			}
+			zoneNameBuffer[trimLength] = '\0';
 		}
 
 		return 0;
```

Comment:
Coding standards suggest to avoid multiple return points. There are two return points here. Can we do a minor cleanup of return statements?

Ref: https://github.com/eclipse/omr/blob/master/doc/CodingStandard.md#return

Reply:
I don't think that advice will improve the readability of this code.